### PR TITLE
updates to directions and test from lecture

### PR DESCRIPTION
**Pull request by adamhinckley**

### CODE REVIEW

File: arrays/arrays.c [hunk 1]
```diff
@@ -73,7 +73,7 @@ void resize_array(Array *arr)
 char *arr_read(Array *arr, int index)
 {
 
-  // Throw an error if the index is greater than the current count
+  // Throw an error if the index is greater than or equal to the current count
```
Comment (tgreenhalgh) [inline arrays/arrays.c]: lines 31-2
```
// Allocate memory for elements
arr->elements = malloc(sizeof(char *) * capacity);
```

you do not want to use malloc per the test ` mu_assert(arr_read(arr, 0) == NULL, "Value initialized not null");`

**Comment by adamhinckley** [comment]
Brady gave that code to us in lecture yesterday. . .

File: arrays/arrays.c (hunk 1 above)
Comment (adamhinckley) [reply arrays/arrays.c]: Brady gave that code to us in lecture yesterday. . .

**Comment by tgreenhalgh** [comment]
in destroy_array, you also need to free up each element
```
for (int i = 0 ; i < arr->count ; i++) {
    free(arr->elements[i]);
  }
```

```
 if (arr->count <= index)
  {
    printf("index is higher than current count");
    return NULL;
  }
  // Otherwise, return the element at the given index
  else
  {
    return arr->elements[index];
  }
```
when you have a return statement in an if, you don't need else because the after the return, it can't go to the `else` - this makes for cleaner code
```
 if (arr->count <= index)
  {
    printf("index is higher than current count");
    return NULL;
  }
// Otherwise, return the element at the given index
return arr->elements[index];

```

In insert, you should return from the error
```
if (arr->count < index)
  {
    printf("index is higher than current count");
    return;
    // exit(1);
  }
```

In arr_append, you are duplicating effort. How could you simplify all that code with just `arr_insert`?

**Pull request merged**